packages/application-shell/src/components/project-container/project-container.js

```diff
@@ -112,9 +124,32 @@ export class ProjectContainer extends React.Component {
     if (this.state.hasError) {
       return <ErrorApologizer />;
     }
-    // TODO: do something if there is an `error`?
-    if (this.props.user && this.props.user.projects.total === 0)
+
+    /**
+     * Given the user does not have any projects and account creation (sign up) is not yet
+     * enabled the user will be logged out.
+     *
+     * Given the user does not have project (and as a result is not part of an organization)
+     * the account application gets control over render. If any other application
+     * is requested to render a full page redirect (to have the proxy serve the request) occurs
+     * given the application is served by the proxy.
+     *    Given the application is not served by the proxy we do not perform a redirect as
+     *    otherwise a redirect loop can occur as no application is able to handle the route.
+     */
+    if (
+      this.props.user.projects.total === 0 &&
+      !this.props.isAccountCreationEnabled
+    )
       return <Redirect to="/logout?reason=no-projects" />;
+    if (this.props.user && this.props.user.projects.total === 0)
+      return (
+        <Switch>
+          <Route path="/account" render={this.props.render} />
+          {this.props.environment.servedByProxy && (
```

Comment:
Hmm that's not what I had in mind. We should still render the `RedirectToProjectCreate` but instead of rendering `null`, we should render a message explaining that there are no projects and that you would need the account app blabla. And of course the redirect should be prevented.

We can also have another component and render one or the other. But the point is to keep the route, otherwise things will probably break.

Reply:
K